docs/entities/livingentity.md

```diff
@@ -0,0 +1,278 @@
+
+- `PathfinderMob`: Contains (surprise!) logic for pathfinding.
+- `AgeableMob`: Contains the logic for aging and baby entities. Zombies and other monsters with baby variants do not extend this class, they instead are children of `Monster`.
+- `Animal`: What most animals extend. Has further abstract subclasses, such as `AbstractHorse` or `TamableAnimal`.
+- `Monster`: The abstract class for most entities the game considers monsters. Like `Animal`, this has further abstract subclasses, such as `AbstractPiglin`, `AbstractSkeleton`, `Raider`, and `Zombie`.
+- `WaterAnimal`: The abstract class for water-based animals, such as fish, squids and dolphins. These are kept separate from the other animals due to significantly different pathfinding.
+
+### Hierarchy of `Player`
+
+Depending on which side the player is on, a different player class is used. You should never need to construct a player yourself, except for `FakePlayer`s.
+
+```mermaid
+graph LR;
+    Player-->AbstractClientPlayer;
+    AbstractClientPlayer-->LocalPlayer;
+    AbstractClientPlayer-->RemotePlayer;
+    Player-->ServerPlayer;
+    ServerPlayer-->FakePlayer;
+    
+    class Player,AbstractClientPlayer red;
+    class LocalPlayer,RemotePlayer,ServerPlayer,FakePlayer blue;
+```
+
+- `AbstractClientPlayer`: This class is used as a base for the two client players, both used to represent players on the [logical client][logicalsides].
+- `LocalPlayer`: This class is used to represent the player currently running the game.
+- `RemotePlayer`: This class is used to represent other players that the `LocalPlayer` may encounter during multiplayer. As such, `RemotePlayer`s do not exist in singleplayer contexts.
+- `ServerPlayer`: This class is used to represent players on the [logical server][logicalsides].
+- `FakePlayer`: This is a special subclass of `ServerPlayer` designed to be used as a mock for a player, for non-player mechanisms that need a player context.
+
+## Spawning
+
+In addition to the [regular ways of spawning][spawning], `Mob`s can also be spawned through some other means. `ArmorStand`s can be spawned through regular means, and `Player`s should not be instantiated yourself, except for `FakePlayer`s.
+
+### Spawn Eggs
+
+It is common (though not required) to [register] a spawn egg for mobs. This is done through the `SpawnEggItem` class, which has been patched by NeoForge to do some extra setup, such as registering the color handler and adding the spawn egg to the internal `SpawnEggItem` -> `EntityType` map.
+
+```java
+// Assume we have a DeferredRegister.Items called ITEMS
+DeferredItem<SpawnEggItem> MY_ENTITY_SPAWN_EGG = ITEMS.registerItem("my_entity_spawn_egg",
+    properties -> new SpawnEggItem(
+        // The entity type to spawn.
+        MY_ENTITY_TYPE.get(),
+        // The colors to use for the spawn egg. The first one is the base/background color,
+        // the second one is the spots/highlight color.
+        0xff0000,
+        0x00ff00,
```

Comment:
These are no longer set on the spawn egg item, but instead as constant tint sources on the client item.